var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -196,13 +196,16 @@ class Hdf5(CMakePackage):
 
     depends_on("cmake@3.12:", type="build")
 
+    depends_on("msmpi", when="+mpi platform=windows")
     depends_on("mpi", when="+mpi")
     depends_on("java", type=("build", "run"), when="+java")
     depends_on("szip", when="+szip")
     depends_on("zlib@1.1.2:")
 
     # The compiler wrappers (h5cc, h5fc, etc.) run 'pkg-config'.
-    depends_on("pkgconfig", type="run")
+    # Skip this on Windows since pkgconfig is autotools
+    for plat in ["cray", "darwin", "linux"]:
+        depends_on("pkgconfig", when="platform=%s" % plat, type="run")
```

Comment:
I'm late with my comments but nonetheless.
1. I disagree with the comment: `pkgconfig` is not `autotools`.
2. The compiler wrappers will not work without `pkg-config` in the `PATH`.

I understand that the compiler wrappers of `hdf5` a rarely used but if `pkg-config` works on Windows, the code above should be reverted. Otherwise, the wrappers should be removed after the installation to avoid confusion.